### ISSUE TITLE
Is it possible to add annotation to an existing annotation?

### ISSUE THREAD

**Issue by superxiao** [manifold-systems/manifold#611]
My use case is I would like to annotate jackson's `JsonProperty` with ErrorProne's [`Keep`](https://errorprone.info/bugpattern/UnusedVariable), so that any field annotated by `JsonProperty` is ignored by ErrorProne's `UnusedVariable` check. Is that possible to do with `@Extension`?

**Comment by superxiao** [comment]
Thank you! To make sure I'm on the same page, I'm wondering about something like the following:

`package ai.extensions.com.fasterxml.jackson.annotation.JsonProperty;

import com.google.errorprone.annotations.Keep;
import manifold.ext.rt.api.Extension;

@Extension
@Keep
public class JsonPropertyExtension {}
`
Which amounts to adding `@Keep` on all fields that are already annotated by `JsonProperty`. Is this what you meant by adding annotations to fields?

**Comment by rsmckinney** [comment]
Ah, I misunderstood. I did not anticipate this use-case, however, so there is a small change I need to make to enable it. I'll have this ready in the next release, which is scheduled for later this evening. Thanks for reporting this.

**Comment by superxiao** [comment]
Wow! Thanks for the fast turnaround. I personally think of this as more of a new feature than a bug fix.

I'm currently using Manifold partly to handle ErrorProne false positives - `ReturnValueIgnored` for things like [Optional.orThrow](https://community.sonarsource.com/t/fp-java-s2201-optional-orelsethrow-return-value-must-be-used/85204), reimplemented as a void version using Manifold. Similarly, to add a `void reportExceptions` to `CompletableFuture`. And now automatically applying `Keep` is possible!

The decision to adopt Manifold was made before I joined my current work and I'm really feeling this is a great choice.

**Comment by superxiao** [comment]
After trying out the the above code piece using manifold 2024.1.24, the compile error on `JsonProperty.java` itself is gone, but the following compile error appears on a pojo class that uses `@JsonProperty` (which also appears in 2024.1.23):

> error: annotation @JsonProperty is missing default values for elements value,namespace,required,index,defaultValue,access

**Comment by rsmckinney** [comment]
Yes, fixing this is part of the change required to support extending annotations. Unfortunately, maven central has not been responsive for most of the evening, which has delayed the release until some time tomorrow. Apologies for the delay.

**Comment by rsmckinney** [comment]
Fix available with release 2024.1.28

**Comment by superxiao** [comment]
Thank you!

**Comment by superxiao** [comment]
@rsmckinney I just tested 2024.1.28, I found that when the extended annotation is on a method it works fine, but when it's on a field (like `JsonProperty` often do) it emits a compile error. I suppose this would need to wait for field annotation support by manifold?

**Comment by rsmckinney** [comment]
> but when it's on a field (like JsonProperty often do) it emits a compile error

It shouldn’t matter because the extension is in the JsonProperty annotation. Can you provide the stack trace and/or compiler error?

**Comment by superxiao** [comment]
Here's a minimal repro project:

[untitled.zip](https://github.com/user-attachments/files/16398536/untitled.zip)

The error I'm seeing:

```
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /com/fasterxml/jackson/annotation/JsonProperty.java:[17,89] cannot find symbol
  symbol:   variable AUTO
  location: @interface com.fasterxml.jackson.annotation.JsonProperty
[INFO] 1 error
[INFO] -------------------------------------------------------------
```

**Comment by rsmckinney** [comment]
Fix available with release 2024.1.29

Thanks again for reporting this, excellent use-case for annotation extensions!
